src/OrchardCore.Modules/OrchardCore.Lucene/Handler/LuceneIndexingContentHandler.cs

```diff
@@ -82,7 +82,8 @@ private static async Task IndexingAsync(ShellScope scope, IEnumerable<ContentCon
 
                 foreach (var indexSettings in await luceneIndexSettingsService.GetSettingsAsync())
                 {
-                    if (indexSettings.IndexedContentTypes.Contains(context.ContentItem.ContentType))
+                    bool ignoreIndexedCulture = indexSettings.Culture == "any" ? false : context.ContentItem.Content?.LocalizationPart?.Culture != indexSettings.Culture;
```

Comment:
Does using the `CultureAspect` achieve this? 

We shouldn't be having to look at `ContentItem.Content` directly like this

Reply:
You mean contentManager.PopulateAspectAsync() ?

Reply:
My code was fine because we need to validate if the content item has a LocalizationPart else calling contentManager.PopulateAspectAsync() will always try to assign a culture to every content item without knowing if it's attached to a LocalizationPart.

Reply:
Else, does the contentManager.PopulateAspectAsync(contentItem, new CultureAspect()) return the proper value all the time? I'm not sure, maybe the issue is in there. I need to take a closer look. For now, what I do here fix the issue explained above.

Reply:
Ok new CultureAspect() sets by default the CultureAspect.Culture to CultureInfo.CurrentUICulture. Here, if we populate the CultureAspect of that content item while wanting to index it, we end up not indexing content items that have no LocalizationPart because it evaluates an inappropriate populated default culture.

Reply:
Ok so culture aspect isn't the right one ;)

So for now I guess is fine, but there is a need a `LocalizationAspect` to return this kind of information.

see this pr too https://github.com/OrchardCMS/OrchardCore/pull/6316

Pop it in as an issue to do later if you prefer

Reply:
I think it should have used Culture aspect. That's exactly what it's made for. The issue is that it still returns a value if no part is attached?  The maybe the aspect should have an extra property (IsDefaultCulture). Because if we have to use LocalizationPart then it makes the aspect useless. And it prevents users from extending it or creating other ways to define localization parts.

Reply:
Let's do this in a different PR because it will take way more time to do. Unit tests and else. I just wanted to fix the indexation for now. I'm pretty sure we have other places that we use the LocalizationPart directly like this.

Reply:
See #7306